File: frontend/src/lib/types/collection.ts
export type CollectionTypes =
	| 'series'
	| 'movies'
	| 'anime'
	| 'manga'
	| 'asian_drama'
	| 'books'
	| '';

export interface CollectionTypeProps {
	name: string;
	value: CollectionTypes;
}

export interface PartialCollectionProps {
	name: string;
	type: CollectionTypes;
	description: string;
	created_at: number;
}

export interface CollectionProps extends PartialCollectionProps {
	id: string;
	key: string;
}

File: api/lib/collection.go
package lib

var CollectionTypes = []string{"series", "movies", "asian_drama", "anime", "books", "manga"}

type CollectionProps struct {
	Name        string `json:"name"`
	ID          string `json:"id,omitempty"`
	Key         string `json:"key,omitempty"`
	Type        string `json:"type"`
	Description string `json:"description"`
	CreatedAt   int    `json:"created_at"`
}
